src/Features/CSharp/Portable/Snippets/CSharpPropgSnippetProvider.cs

```diff
@@ -35,7 +36,7 @@ internal sealed class CSharpPropgSnippetProvider() : AbstractCSharpAutoPropertyS
         // Having a property with `set` accessor in a readonly struct leads to a compiler error.
         // So if user executes snippet inside a readonly struct the right thing to do is to not generate `set` accessor at all
         if (syntaxContext.ContainingTypeDeclaration is StructDeclarationSyntax structDeclaration &&
-            structDeclaration.Modifiers.Any(SyntaxKind.ReadOnlyKeyword))
+            syntaxContext.SemanticModel.GetDeclaredSymbol(structDeclaration, cancellationToken) is { IsReadOnly: true })
```

Comment:
📝 Same comment applies here. You may want to share code in the base class for this?